lib/api/accounts/user-accounts.js

```diff
@@ -0,0 +1,143 @@
+'use strict'
+
+const express = require('express')
+const bodyParser = require('body-parser').urlencoded({ extended: false })
+const debug = require('../../debug').accounts
+
+const CreateAccountRequest = require('../../requests/create-account-request')
+const AddCertificateRequest = require('../../requests/add-cert-request')
+
+/**
+ * Returns an Express middleware handler for checking if a particular account
+ * exists (used by Signup apps).
+ *
+ * @param accountManager {AccountManager}
+ *
+ * @return {Function}
+ */
+function checkAccountExists (accountManager) {
+  return (req, res, next) => {
+    let accountUri = req.hostname
```

Comment:
Not sure what the question is. It's not checking the origin, it's checking the request's hostname. (This is what the existing logic does, as well.)